scripts/deploy/update-npm-tag.js

```diff
@@ -0,0 +1,55 @@
+const {promisify} = require('util');
+const exec = promisify(require('child_process').exec);
+
+const packageDirs = ['atomic', 'bueno', 'headless'];
+
+async function main() {
+  const requests = packageDirs
+    .map(dir => require(`../../packages/${dir}/package.json`))
+    .map(({name, version}) => updateNpmTag(name, version))
+
+  await Promise.all(requests);
+}
+
+async function updateNpmTag(packageName, version) {
+  const tag = process.argv[2];
+  const latestVersion = await getLatestVersion(packageName);
+
+  if (!isGreaterThanLatestVersion(version, latestVersion)) {
+    console.log(`skipping tag update for ${packageName} because version "${version}" is not greater than latest version "${latestVersion}".`)
+    return;
+  }
+  
+  console.log(`updating ${packageName}@${version} to ${tag}.`);
+  await exec(`npm dist-tag add ${packageName}@${version} ${tag}`)
+}
+
+async function getLatestVersion(packageName) {
```

Comment:
The changes are not "instant" on npm registry.

When you publish a new package version, it can take couple minutes for it to propagate in the registry.

Meaning that `npm view @coveo/headless version` could return something that is not the "real" latest version, if it was uploaded couple seconds/minutes ago.

So the automated process might fail, or we'd need to have some kind of mechanism to do something similar to "we are releasing version X, poll npm with a backoff until the version is really available in the registry".

Reply:
Hmm, I don't think a polling mechanism needed. The scenario I want to defend against is an older build lowering the latest tag, creating a sort of wrench.

So I think for the scenario to play out, someone has to a) be in process of releasing to prod around the same time that someone else decides to rerun an old build in prod that is b) greater than the latest, but c) less than than the actual latest version that hasn't been reflected yet.

I don't think it is likely for this to happen, but even if it does, though the wrong version might be marked as latest, both will still be greater than the last version.

Reply:
Ok !